src/Concordium/Client/Types/Transaction.hs

```diff
@@ -297,28 +391,36 @@ data SignedTransaction = SignedTransaction
       stExpiryTime :: !TransactionExpiryTime,
       -- | Account nonce.
       stNonce :: !Nonce,
-      -- | Signer account address.
+      -- | Sender account address.
       stSigner :: !AccountAddress,
+      -- | Sponsor account address.
+      stSponsor :: !(Maybe AccountAddress),
       -- | The payload of the transaction.
       stPayload :: !Types.Payload,
-      -- | Signatures generated by the signer account. This map might contain enough signatures to send the transaction on-chain or
+      -- | Signatures generated for the sender account. This map might contain enough signatures to send the transaction on-chain or
+      -- additional signatures are needed before the transaction is considered fully signed.
+      stSignature :: !Types.TransactionSignature,
+      -- | Signatures generated for the sponsor account. This map might contain enough signatures to send the transaction on-chain or
       -- additional signatures are needed before the transaction is considered fully signed.
-      stSignature :: !Types.TransactionSignature
+      stSignatureSponsor :: !(Maybe Types.TransactionSignature)
```

Comment:
I think this is potentially insufficient, because there is no indication if a non-sponsored transaction is V0 or V0, which affects the hashing and eventual serialization. Perhaps have an `stExtended ;: !Bool`, which defaults to `False` when deserializing (for backwards compatibility), but we should require that `stSponsor` and `stSignatureSponsor` are `Nothing` if `stExtended` is `False`.

Reply:
Added in added in [0074a37](https://github.com/Concordium/concordium-client/pull/404/commits/0074a371e16b4401eeeeea4636c07fb82e37379e) on #404